AzureAttestationExtension/agent.py

```diff
@@ -0,0 +1,708 @@
+#!/usr/bin/env python
+#
+# AzureAttestationLinuxAgent Extension
+#
+# Copyright 2020 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import os.path
+import signal
+import pwd
+import grp
+import re
+import sys
+import traceback
+import time
+import platform
+import subprocess
+import json
+import base64
+import inspect
+import shutil
+import crypt
+import xml.dom.minidom
+import re
+from distutils.version import LooseVersion
+
+from threading import Thread
+
+try:
+    from Utils.WAAgentUtil import waagent
+    import Utils.HandlerUtil as HUtil    
+    
+except Exception as e:
+    # These utils have checks around the use of them; this is not an exit case
+    print('Importing utils failed with error: {0}'.format(e))
```

Comment:
Incase WAAgentUtil module couldn't be imported we should log and throw or exit from here. At some places, waagent is used without checking and as the first instance, at init_waagent_logger, it will throw the undefined symbols. (Also a misleading error)

Reply:
I will change other places where we are using waagent without checking.